modules/programs/zellij.nix

```diff
@@ -69,17 +107,44 @@ in {
         text = lib.hm.generators.toKDL { } cfg.settings;
       };
 
-    programs.bash.initExtra = mkIf cfg.enableBashIntegration (mkOrder 200 ''
-      eval "$(${zellijCmd} setup --generate-auto-start bash)"
-    '');
-
-    programs.zsh.initExtra = mkIf cfg.enableZshIntegration (mkOrder 200 ''
-      eval "$(${zellijCmd} setup --generate-auto-start zsh)"
-    '');
-
-    programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration
-      (mkOrder 200 ''
+    programs.zsh.initExtra = mkIf (cfg.enableZshIntegration) (mkOrder 200
+      ((if cfg.autostartOnShellStart.enable then ''
+        eval "$(${zellijCmd} setup --generate-auto-start zsh)"
+      '' else
+        "") + (if cfg.enableShellCompletions then ''
+          # TMPFIX: Until Zellij fixes completions generation and functions creation, a workaround is necessary.
+          #  Used workaround: https://github.com/zellij-org/zellij/issues/1933#issuecomment-2274464004.
+          autoload -U +X compinit && compinit
+          source <(${zellijCmd} setup --generate-completion zsh | sed -Ee 's/^(_(zellij) ).*/compdef \1\2/')
+        '' else
+          "")));
+
+    programs.fish.interactiveShellInit = mkIf (cfg.enableFishIntegration)
+      (mkOrder 200 ((if cfg.autostartOnShellStart.enable then ''
         eval (${zellijCmd} setup --generate-auto-start fish | string collect)
-      '');
+      '' else
+        "") + (if cfg.enableShellCompletions then ''
+          eval (${zellijCmd} setup --generate-completion fish | string collect)
+        '' else
+          "")));
+
+    programs.bash.initExtra = mkIf (cfg.enableBashIntegration) (mkOrder 200
```

Comment:
Why exactly is the mkOrder 200 required?

Reply:
It is not required. Some packages in hm use this to place the generated completions/aliases a bit earlier in the config files for each respective shell so that the generated options can be easily modified by manual configuration or other programs. However, it is not necessary at all and I can remove them.